Infrastructure.DataAccess/Migrations/SQLScripts/Deactive_Expired_Advice_And_Schedule_Missing_Jobs.sql

```diff
@@ -0,0 +1,217 @@
+/*
+User story reference: 
+    https://os2web.atlassian.net/browse/KITOSUDV-2243
+ 
+Content:
+	- Removes old Hangfire jobs which have no related data and should be deleted.
+	- Updates IsActive column on Advice, which should be deactivated when Hangfire runs next send. But won't be as there is no Hangfire job
+	- Recreates Hangfire jobs on Advices where there should be a job but isn't one.
+*/
+
+/* Note: Check to see if Hangfire tables exists so that script doesn't run on fresh install */
+IF OBJECT_ID('kitos_HangfireDB.HangFire.Set', 'U') IS NOT NULL AND 
+	OBJECT_ID('kitos_HangfireDB.HangFire.Hash', 'U') IS NOT NULL AND 
+	OBJECT_ID('kitos_HangfireDB.HangFire.Job', 'U') IS NOT NULL AND 
+	OBJECT_ID('kitos_HangfireDB.HangFire.JobParameter', 'U') IS NOT NULL
+
+	BEGIN
+		/* Part 1: Delete recurring jobs with no related data from hangfire DB to clean up*/
+		DECLARE @RecurringJobsWithData TABLE 
+		(
+			AdviceId int
+		)
+
+		INSERT INTO @RecurringJobsWithData
+		SELECT CONVERT(int, SUBSTRING([Key], 23, 28)) AS AdviceId
+		FROM [kitos_HangfireDB].[HangFire].[Hash]
+		WHERE [Key] LIKE '%recurring-job:Advice: %'
+
+		INSERT INTO @RecurringJobsWithData
+		SELECT CONVERT(int, REVERSE(SUBSTRING(REVERSE(SUBSTRING([Arguments], 3, 8)), 3, 8))) AS AdviceId
+		FROM [kitos_HangfireDB].[HangFire].[Job]
+		WHERE [Arguments] NOT LIKE '%null%'
+
+
+		DECLARE @RecurringAdviceJobs TABLE 
+		(
+			AdviceId int,
+			SetId int
+		)
+
+		INSERT INTO @RecurringAdviceJobs
+		SELECT CONVERT(int, SUBSTRING([Value], 8, 6)), Id
+		FROM [kitos_HangfireDB].[HangFire].[Set]
+		WHERE [Key] LIKE 'recurring-jobs' AND [Score] > 0 AND [Value] LIKE 'Advice: %'
+
+		DELETE
+		FROM [kitos_HangfireDB].[HangFire].[Set] 
+		WHERE Id IN (
+			SELECT SetId
+			FROM @RecurringAdviceJobs
+			WHERE AdviceId NOT IN (
+				SELECT * 
+				FROM @RecurringJobsWithData
+			)
+		)
+
+
+		/* Part 2: Find Id's of Advice which have related hangfire data */
```

Comment:
det er vel de udløbne advices uden related hf data

Reply:
jf ` Id not in`

Reply:
Nej her finder vi faktisk de advice som har HF data. Og det bruger vi så til at vide hvilke Advice vi ikke skal røre ved